src/compat/posix/fs/dvfs/libgen.c

```diff
@@ -0,0 +1,79 @@
+/**
+ * @file
+ *
+ * @date 28.10.13
+ * @author Anton Bondarev
+ */
+
+#include <limits.h>
+#include <string.h>
+#include <libgen.h>
+
+static char *path_final_slash(char *path) {
+	size_t path_len = strlen(path);
+	char *c;
+
+	c = path + path_len - 1;
+	while (*c == '/') {
+		*c-- = '\0';
+		path_len--;
+	}
+
+	while (path_len > 0 && *c != '/') {
+		c--;
+		path_len--;
+	}
+
+	if (path_len > 0) {
+		return c;
+	}
+
+	return NULL;
+}
+
+char * basename(char *path) {
+	static char bname_buff[2];
```

Comment:
is this variable used  only for return "."? Why don't use it for "/" or better remove one and use return constant string 
return "."